controller/events.go

```diff
@@ -177,6 +370,13 @@ func (e *EventListener) Notify(event *ct.AppEvent) {
 			sub.Notify(event)
 		}
 	}
+	if event.AppID != "" {
```

Comment:
It's to handle the case where the subscriber is in `subscribers[""]` and `event.AppID` is set, e.g. subscribing to all app events but not specifying an object ID.